cmd/tag.go

```diff
@@ -0,0 +1,146 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	tagAddArgNum       int = 2
+	tagClearArgNum     int = 1
+	tagListArgNum      int = 1
+	tagRemoveMinArgNum int = 2
+)
+
+func tagCmd() *cobra.Command {
+	tagCmd := &cobra.Command{
+		Use:   "tag",
+		Short: "Edit the tags of a document",
+		Long:  "Edit the tags of a document",
+	}
+
+	tagCmd.AddCommand(tagAddCmd(), tagClearCmd(), tagListCmd(), tagRemoveCmd())
+
+	return tagCmd
+}
+
+func tagAddCmd() *cobra.Command {
+	addCmd := &cobra.Command{
+		Use:   "add [flags] SBOM_ID TAGS...",
+		Short: "Add tags to a document",
+		Long:  "Add tags to a document",
+		Args:  cobra.MinimumNArgs(tagAddArgNum),
+		Run: func(cmd *cobra.Command, args []string) {
+			backend := backendFromContext(cmd)
+
+			defer backend.CloseClient()
+
+			document, err := backend.GetDocumentByIDOrAlias(args[0])
+			if err != nil {
+				backend.Logger.Fatalf("failed to get document: %v", err)
+			}
+
+			if err := backend.AddAnnotations(document.Metadata.Id, "tag", args[1:]...); err != nil {
+				backend.Logger.Fatalf("failed to add tags: %v", err)
+			}
+		},
+	}
+
+	return addCmd
+}
+
+func tagClearCmd() *cobra.Command {
+	clearCmd := &cobra.Command{
+		Use:   "clear [flags] SBOM_ID...",
+		Short: "Clear the tags of a document",
+		Long:  "Clear the tags of a document",
+		Args:  cobra.ExactArgs(tagClearArgNum),
+		Run: func(cmd *cobra.Command, args []string) {
+			backend := backendFromContext(cmd)
+
+			defer backend.CloseClient()
+
+			document, err := backend.GetDocumentByIDOrAlias(args[0])
+			if err != nil {
+				backend.Logger.Fatalf("failed to get document: %v", err)
+			}
+
+			annotationsToRemove, err := backend.GetDocumentAnnotations(document.Metadata.Id, "tag")
+			if err != nil {
+				backend.Logger.Fatalf("failed to clear tags: %v", err)
+			}
+
+			tagsToRemove := make([]string, len(annotationsToRemove))
+			for i, annotation := range annotationsToRemove {
+				tagsToRemove[i] = annotation.Value
+			}
```

Comment:
If looping and assigning this way, prefer to see

- rename `i` ---> `idx` to align with project convention
- just use `for idx := range annotationsToRemove` and reference as `annotationsToRemove[idx].Value`

Reply:
Also can you describe the optimization this implements, specifically?